internal/testrunner/runners/system/servicedeployer/kubernetes.go

```diff
@@ -144,14 +151,84 @@ func findKubernetesDefinitions(definitionsDir string) ([]string, error) {
+func downloadElasticAgentManagedYAML(url string) ([]byte, error) {
+	// Get the data
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get file from URL %s", url)
+	}
+	defer resp.Body.Close()
+	logger.Debugf("status code when downloading elastic-agent-managed-kubernetes.yaml is %d", resp.StatusCode)
+	if resp.StatusCode != 200 {
+		return nil, errors.Wrapf(err, "downloading failed due to status code %d", resp.StatusCode)
+	}
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read response body")
+	}
+	return b, nil
+}
+
+// getElasticAgentYAML retrieves elastic-agent-managed.yaml from upstream and modifies the file as needed
+// to run locally.
+func getElasticAgentYAML() ([]byte, error) {
+	appConfig, err := install.Configuration()
+	if err != nil {
+		return nil, errors.Wrap(err, "can't read application configuration")
+	}
+
+	logger.Debugf("downloading elastic-agent-managed-kubernetes.yaml from %s", elasticAgentManagedYamlURL)
+	// retry downloading elastic agent manifest for 5 times (sleep 10 seconds between each try) in case of error
+	elasticAgentManagedYaml, err := retryDownloadElasticAgentManagedYAML(elasticAgentManagedYamlURL, 5, 10, downloadElasticAgentManagedYAML)
+	if err != nil {
+		return nil, errors.Wrapf(err, "downloading failed for file from source  %s", elasticAgentManagedYamlURL)
+	}
+
+	// Set regex to match fleet url from yaml file
+	fleetURLRegex := regexp.MustCompile("http(s){0,1}:\\/\\/fleet-server:(\\d+)")
+	// Replace fleet url
+	elasticAgentManagedYaml = fleetURLRegex.ReplaceAll(elasticAgentManagedYaml, []byte("http://fleet-server:8220"))
+
+	// Set regex to match image name from yaml file
+	imageRegex := regexp.MustCompile("docker.elastic.co/beats/elastic-agent:\\d.+")
+	// Replace image name
+	elasticAgentManagedYaml = imageRegex.ReplaceAll(elasticAgentManagedYaml, []byte(appConfig.DefaultStackImageRefs().ElasticAgent))
+
+	return elasticAgentManagedYaml, nil
+}
+
+// retryDownloadElasticAgentManagedYAML retries downloading elastic agent managed manifest for x attempts
+// until there is no error and bytes of the file are more than 2000.
+func retryDownloadElasticAgentManagedYAML(url string, attempts int, sleep time.Duration, f func(string) ([]byte, error)) ([]byte, error) {
+	var err error
+	for i := 0; i < attempts; i++ {
+		if i > 0 {
+			logger.Debugf("retrying download after error: %s", err)
+			time.Sleep(sleep)
+		}
+		elasticAgentManagedYaml, err := f(url)
+		if err == nil {
+			logger.Debugf("downloaded %d bytes", len(elasticAgentManagedYaml))
+			if len(elasticAgentManagedYaml) > 2000 {
+				return elasticAgentManagedYaml, nil
+			}
+			err = fmt.Errorf("bytes downloaded should be more than 2000 but where: %d", len(elasticAgentManagedYaml))
+			logger.Debugf("failed because of %s", err)
+			logger.Debugf("file downloaded is %s", string(elasticAgentManagedYaml))
```

Comment:
I discourage printing the doc here. If you really want to check if the content is right, maybe print just MD5?